frontend/src/components/common/StateHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import type { ApolloError } from "@apollo/client";
+
+import ErrorState from "./info/ErrorState";
+import LoadingState from "./info/LoadingState";
+
+interface StateHandlerProps {
+  loading: boolean;
+  error?: ApolloError;
+  isEmpty?: boolean;
+  emptyState?: React.ReactNode;
+  children: React.ReactNode | React.ReactNode[];
+  fullPage?: boolean;
+}
+
+const StateHandler = ({
+  loading,
+  error,
+  isEmpty = false,
+  emptyState,
+  children,
+  fullPage,
+}: StateHandlerProps): React.ReactElement => {
+  if (loading) return <LoadingState fullPage={fullPage} />;
+
+  if (!!error) return <ErrorState fullPage={fullPage} />;
+
+  if (isEmpty && emptyState) return <>{emptyState}</>;
```

Comment:
It strikes me that most of our components won't use this prop, and those that do have very simple logic. I'm thinking it might be best to remove this prop in favour of components handling their own empty states, since this component doesn't really save any work for the caller right now